File: Ninja Slayground 2.0/12. Add One To Number/Solution.py
#Learn the code, Copy and Paste is not allowed. You are required to type...

#Follow this account, for all future code updates on GitHub.

from os import *
from sys import *
from collections import *
from math import *

def addOneToNumber(arr):
    #   Write your code here
    n = len(arr)
    carry = 1
    for i in range(n - 1, -1, -1):
        arr[i] += carry
        if arr[i] > 9:
            arr[i] = 0
            carry = 1
        else:
            carry = 0
            break
    if carry:
        arr.insert(0, 1)
    while arr[0] == 0:
        arr.pop(0)
    return arr

File: Ninja Slayground 2.0/12. Add One To Number/Solution.java
// Learn the code, And copy and paste is not allowed. You are required to type...

// Follow this account, for all future code updates on GitHub.

import java.util.* ;
import java.io.*; 

public class Solution
{
    public static ArrayList<Integer> addOneToNumber(ArrayList<Integer> arr)
    {
        int n = arr.size();
        int carry = 1;

        for (int i = n - 1; i >= 0; i--) {
            arr.set(i, arr.get(i) + carry);
            if (arr.get(i) > 9) {
                arr.set(i, 0);
                carry = 1;
            } else {
                carry = 0;
                break;
            }
        }

        if (carry == 1) {
            arr.add(0, 1);
        }

        while (arr.size() > 0 && arr.get(0) == 0) {
            arr.remove(0);
        }

        return arr;
    }
}
